### ISSUE TITLE
Data Protection Act (RGPD) in France

### ISSUE THREAD

**Issue by etsol97120** [openNDS/openNDS#62]
Hello, I am setting up a public wifi hotspot using NDS and I would like some informations. In France, the law requires us to store the sites that the customer has visited as well as his connection time. Is there a variable allowing to know the connection time of the cleint as well as to retrieve the sites it has visited?

**Comment by sicelo** [comment]
> Is there a variable allowing to know the connection time of the cleint

Yes. For example, in the deauth section of `binauth_log`, you can see that `session_start` and `session_end` are returned for that client

**Comment by bluewavenet** [comment]
@etsol97120 

**openNDS does not support access logging.**

For your information:

In each member state of the European Union (EU) there are two conflicting laws that apply to providing a public wifi service.
These are:
1. General Data Protection Regulation (GDPR or RGPD in French)
2. Local EU nation state replacements for the European Court of Justice (ECJ) revoked Data Retention Directive (DRD)

Both of these are outside the scope of openNDS, but compliance is supported via the FAS API.

The conflict arises when a local Data Retention Law requires collection of personal data, yet the GDPR _requires_ comprehensive protection of personal data.
Personal data must not be stored unless the consent of the individual is obtained (usually by being given access to and then accepting the Terms of Service for the portal).

In summary, any personal data your captive portal collects (including sites visited), other than that required for secure login and anonymous statistics gathering must by EU law be stored in an _encrypted_ form at a _secure location_ and be accessible only with:
1. The consent of the individual to which the data refers; or
2. The issue of a court warrant on behalf of a relevant body (ie police, national security agency etc)

**Comment by etsol97120** [comment]
@sicelo : Hello, thank you for your feedback. Indeed by retrieving the date and time I could define the connection time for each MAC address. I will go through a PHP page in order to return this information to a database. Thanks again.

@bluewavenet : You're absolutely right. I have just reviewed the directives indicated in the RGPD and the sites visited are not requested. It is true that French and European laws contradict each other. I will have to define the clauses carefully. The collected data will be sent to a secure database. They must be stored for a period of one year so that if a forensic investigation is opened, I can forward them to the authorities. Users will have to accept the clauses that I will have established in order to comply with the law.
